app/src/main/kotlin/it/vfsfitvnm/vimusic/ui/screens/BuiltInPlaylistScreen.kt

```diff
@@ -71,13 +71,7 @@ fun BuiltInPlaylistScreen(builtInPlaylist: BuiltInPlaylist) {
             val songs by remember(context.globalCache, builtInPlaylist) {
                 when (builtInPlaylist) {
                     BuiltInPlaylist.Favorites -> Database.favorites()
-                    BuiltInPlaylist.Offline -> Database.songsWithContentLength().map { songs ->
-                        songs.filter { song ->
-                            song.contentLength?.let {
-                                context.globalCache.isCached(song.id, 0, song.contentLength)
-                            } ?: false
-                        }
-                    }
+                    BuiltInPlaylist.Offline -> Database.getDownloadedSongs()
```

Comment:
let's still have `.filter { song ->
                                context.globalCache.isCached(song.id, 0, song.contentLength)
                        }`
to double check that the songs are cached